### PR TITLE
When checking deltemplate compatibility, only check required params.

### PR DESCRIPTION
**Pull request by nicks**
Fixes https://github.com/google/closure-templates/issues/33

### CODE REVIEW

**Comment by nicks** [comment]
the cla googlebot is pretty cool, btw

**Comment by Ubehebe** [comment]
Damn, I just edited CheckDelegatesVisitor today. I'll try to do another GitHub push on Monday so you can rebase this if need be.

**Comment by nicks** [comment]
ping?

**Comment by Ubehebe** [comment]
@nicks Pushed. Rebase?

**Comment by nicks** [comment]
rebased!

File: java/src/com/google/template/soy/parsepasses/CheckDelegatesVisitor.java [hunk 1]
```diff
@@ -171,6 +172,17 @@ private void checkTemplates() {
   }
 
 
+  private Set<TemplateParam> getRequiredParamSet(TemplateDelegateNode delTemplate) {
```
Comment (Ubehebe) [inline java/src/com/google/template/soy/parsepasses/CheckDelegatesVisitor.java]: static?

Comment (nicks) [inline java/src/com/google/template/soy/parsepasses/CheckDelegatesVisitor.java]: done!

**Comment by Ubehebe** [comment]
@nicks Rebase one more time, sorry. I'll try to merge this by the end of the day if you can turn this around.

**Comment by nicks** [comment]
rebased! 

rebasing isn't that hard tho, it's usually imports. it'd probably be easier to just merge yourself with git-cherry-pick?

**Comment by Ubehebe** [comment]
@nicks Currently running internal tests on the patch.

**Comment by nicks** [comment]
wooooooooooo! thanks brendan.

**Pull request merged**